### DISCUSSION TITLE
Duplicate roles globally and unique for each company | Multitenancy

### DISCUSSION THREAD

**Discussion by Luissylar** [spatie/laravel-permission#2773] [General]
Hello, I have been creating a multi-tenancy system for companies where each company has its roles, each company creates roles grouping permissions at will, the problem is that a company could create a role with the same name as another company and that is where it is generated the duplication error, as it would be that the roles at a global level can be duplicated, but at the company level there is no duplication

here it indicates that the company has 0 roles, however another company has the same role, so it does not allow registration
![imagen](https://github.com/user-attachments/assets/55a7ac28-8456-4dd2-ab85-cb06e34dc77f)

I also tried to modify the migrations, but apparently it is fine since it verifies with the company ID, This is permissions migration install with package

```php
 Schema::create($tableNames['roles'], function (Blueprint $table) use ($teams, $columnNames) {
            //$table->engine('InnoDB');
            $table->bigIncrements('id'); // role id
            if ($teams || config('permission.testing')) { // permission.testing is a fix for sqlite testing
                $table->unsignedBigInteger($columnNames['team_foreign_key'])->nullable();
                $table->index($columnNames['team_foreign_key'], 'roles_team_foreign_key_index');
            }
            $table->string('name');       // For MyISAM use string('name', 225); // (or 166 for InnoDB with Redundant/Compact row format)
            $table->string('guard_name'); // For MyISAM use string('guard_name', 25);
            $table->timestamps();
            if ($teams || config('permission.testing')) {
                $table->unique([$columnNames['team_foreign_key'], 'name', 'guard_name']);
            } else {
                $table->unique(['name', 'guard_name']);
            }
        });

**Comment by drbyte** [comment]
> creating a multi-tenancy system

Are you using the Teams feature of this package?

**Reply by Luissylar** [reply]
Apologies for the case, I solved it by putting a validation, in the input validating as a single field along with company_id, I am using Laravel filament with the Companies plugin, with this each company can have the same roles at a global level but without repeating themselves at the level of possession.

add unique with validation rule 

```php
Forms\Components\TextInput::make('name')
                                    ->label(__('filament-shield::filament-shield.field.name'))
                                    ->unique(
                                        modifyRuleUsing: function (\Illuminate\Validation\Rules\Unique $rule) {
                                            return $rule->where('company_id', auth()->user()->currentCompany->id ?? null);
                                        },
                                        ignoreRecord: true
                                    )
                                    ->maxLength(255),
